app/src/features/rules/screens/rulesList/components/RulesList/components/RulesListContentHeader/RulesListContentHeader.tsx

```diff
@@ -187,7 +189,7 @@ const RulesListContentHeader: React.FC<Props> = ({ searchValue, setSearchValue,
     <ContentListHeader
       title="My Rules"
       subtitle="Create and manage your rules from here"
-      actions={contentHeaderActions}
+      actions={userRole !== TeamRole.read ? contentHeaderActions : null}
```

Comment:
just a nit, I feel this is easier to read as:

```suggestion
      actions={userRole === TeamRole.read ? null : contentHeaderActions}
```